src/auth.js

```diff
@@ -0,0 +1,54 @@
+import jwt from 'jsonwebtoken'
+
+export async function login(request, response) {
+    try {
+        //Get username
+        const {username} = request.body
+
+        //If no username this doesnt run
+        if(username) {
+
+            //just giving id a value 
+            const id = new Date().getDate()
+
+            //creating a token that is valid for 15 mins
+            const token = jwt.sign({id, username}, process.env.JWT_SECRET, {expiresIn:'15min'}) 
+
+            //set cookie with token
+            response.cookie('jwt', token, {httpOnly: true, maxAge: 15 * 60 * 1000})
```

Comment:
The use of `httpOnly` is great, because it means that the cookie will only be available to the server, never to any frontend javascript code. This is important for security. Without it, an XSS attack on your website could steal users' cookies, which it can't now.